File: front/src/app/models/Url.ts
export class Url {
   public id?: number;
   public shortUrl?: string;
   public targetUrl?: string;
   public password?: boolean;
   public createdDate?: string;

   constructor(id?: number, shortUrl?: string, targetUrl?: string, password?: boolean, createdDate?: string) {
      this.id = id;
      this.shortUrl = shortUrl;
      this.targetUrl = targetUrl;
      this.password = password;
      this.createdDate = createdDate;
   }
}

File: back/UrlShortener/Models/Url.cs
namespace UrlShortener.Models;

using System;
using System.ComponentModel.DataAnnotations.Schema;

[Table("Url")]
public class Url
{
   public int Id { get; set; }

   public string ShortUrl { get; set; }

   public string TargetUrl { get; set; }

   public DateTime CreatedDate { get; set; }

   public string Password { get; set; }
}
